src/main/java/com/recurse/portfolio/web/ProjectController.java

```diff
@@ -79,29 +81,69 @@ public ModelAndView postNewProject(
                     tag.getTagId()
                 );
             }
+            
+            String slugifiedProjectName = slugifyString(savedProject.getName());
+            
             return new ModelAndView(new RedirectView(
-                "/project/" + savedProject.getProjectId()
+                "/project/" + savedProject.getProjectId() + "/" + slugifiedProjectName
             ));
         }
     }
-
-    @GetMapping("/project/{projectId}")
+    
+    // slugify, i.e. create URL-friendly version of, the project name
+    private String slugifyString(String text)
+    {
+        Slugify slugifier = new Slugify();
+        String slugifiedProjectName = slugifier.slugify(text);
+        return slugifiedProjectName;
+    }
```

Comment:
Making a helper method was a good call, but I think we can make this both more performant and shorter:

If we move the `slugifier` to be an instance variable (`private Slugify slugifier = new Slugify();`), then all that's left is `return slugifier.slugify(text);`, which we could inline.